internal/handler/processor/auth_validator.go

```diff
@@ -0,0 +1,152 @@
+	githubController *controllers.GitHub
+}
+
+func NewAuthValidatorProcessor(githubController *controllers.GitHub, opts ...Option) Processor {
+	_inst := &authValidatorProcessor{githubController: githubController, logger: helpers.NewNoopLogger()}
+	applyOpts(_inst, opts...)
+	return _inst
+}
+
+func (p *authValidatorProcessor) SetLogger(logger *slog.Logger) {
+	p.logger = logger.WithGroup("pre-processor:validator")
+}
+
+func (p *authValidatorProcessor) Process(req any) (bus *promotion.Bus, err error) {
+	authRequest, ok := req.(*AuthRequest)
+	if !ok {
+		return nil, promotion.NewInternalError("invalid event type. expected *AuthRequest got %T", req)
+	}
+	body, headers := authRequest.Body, authRequest.Headers
+
+	// Default response
+	bus = &promotion.Bus{
+		Body:     body,
+		Headers:  headers,
+		Response: models.Response{StatusCode: http.StatusUnprocessableEntity},
+	}
+
+	p.logger.Info("processing request...")
+
+	eventType, found := headers[strings.ToLower(github.EventTypeHeader)]
+	if !found {
+		p.logger.Warn("missing event type")
+		return &promotion.Bus{
+			Response: models.Response{Body: "missing event type", StatusCode: http.StatusUnprocessableEntity},
+		}, promotion.NewInternalError("missing event type")
+	}
+	bus.EventType = eventType
+
+	deliveryId, found := headers[strings.ToLower(github.DeliveryIDHeader)]
+	if !found {
+		p.logger.Warn("missing delivery ID")
+		return &promotion.Bus{
+			Response: models.Response{Body: "missing delivery ID", StatusCode: http.StatusUnprocessableEntity},
+		}, promotion.NewInternalError("missing delivery ID")
+	}
+
+	// Validate the request
+	resp, err := p.checkEventType(eventType, authRequest.EventProcessors)
+	if err != nil {
+		p.logger.Warn("validating request", slog.Any("error", err))
+		return &promotion.Bus{
+			Response: *resp,
+		}, promotion.NewInternalError("failed to validate request. error: %v", err)
+	}
+
+	// Add the event type to the logger now that we know it's valid
+	p.logger = p.logger.With(slog.String("event", eventType))
+
+	// Refresh credentials if needed
+	if err = p.githubController.RetrieveCredentials(); err != nil {
+		p.logger.Warn("failed to refresh credentials", slog.Any("error", err))
+		return &promotion.Bus{
+			Response: models.Response{Body: err.Error(), StatusCode: http.StatusUnauthorized},
+		}, promotion.NewInternalError("failed to refresh credentials. error: %v", err)
+	}
+	if err = p.githubController.ValidateWebhookSecret(body, headers); err != nil {
+		p.logger.Warn("validating signature", slog.Any("error", err))
+		return &promotion.Bus{
+			Response: models.Response{Body: err.Error(), StatusCode: http.StatusForbidden},
+		}, promotion.NewInternalError("failed to validate signature. error: %v", err)
+	}
+	p.logger.Debug("request body is valid")
+
+	// Add the delivery ID to the logger, now that we know the payload is valid
+	p.logger = p.logger.With(slog.String("deliveryId", deliveryId))
+
+	repo, err := p.ExtractCommonRepository(body)
```

Comment:
"Common Repository" or "Repository Context" (based on the error messages)?

Reply:
Refactored. :-)